File: 프로그래머스/lv0/120818. 옷가게 할인 받기/옷가게 할인 받기.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int price)
{
    if (price >= 100000 && price < 299999)
    {
        return price -= price * 0.05;
    }
    else if (price >= 300000 && price < 499999)
    {
        return price -= price * 0.1;
    }
    else if (price >= 500000)
    {
        return price -= price * 0.2;
    }
    
    return price;
}

File: 프로그래머스/0/120818. 옷가게 할인 받기/옷가게 할인 받기.java
class Solution {
    public int solution(int price) {
        int discountRate = getDiscountRate(price);
        return price * (100 - discountRate) / 100;
    }
    
    private int getDiscountRate(int price) {
        if (price >= 500_000) return 20;
        if (price >= 300_000) return 10;
        if (price >= 100_000) return 5;
        return 0;
    }
    
}